packages/backend/src/Tabulators/MethodOfEqualShares.ts

```diff
@@ -0,0 +1,561 @@
+  utility: Utility,
+  parsedData: IparsedData,
+): methodOfEqualSharesSummaryData {
+  const totalScores: totalScore[] = candidates.map((candidate) => ({
+    index: candidate.index,
+    score: Object.values(utility[candidate.index.toString()]).reduce(
+      (a, b) => a + b,
+      0,
+    ),
+  }));
+
+  const nBulletVotes = parsedData.scores.reduce((count, vote: ballot) => {
+    const nonZeroScores = vote.filter((score) => score > 0).length;
+    return count + (nonZeroScores === 1 ? 1 : 0);
+  }, 0);
+
+  return {
+    candidates,
+    totalScores,
+    nValidVotes: parsedData.validVotes.length,
+    nInvalidVotes: parsedData.invalidVotes.length,
+    nUnderVotes: parsedData.underVotes,
+    nBulletVotes,
+  };
+}
+
+function breakTies(
+  cost: Cost,
+  totalUtility: { [candidateId: string]: number },
+  choices: string[],
+  tieBreakOrder: string[],
+): string[] {
+  let remaining = choices.slice();
+
+  // Select candidate(s) with lowest cost
+  const bestCost = Math.min(
+    ...remaining.map((candidateId) => cost[candidateId]),
+  );
+  remaining = remaining.filter((candidateId) =>
+    isClose(cost[candidateId], bestCost),
+  );
+
+  if (remaining.length === 1) {
+    return remaining;
+  }
+
+  // If tied on cost, select candidate(s) with highest initial vote count
+  const bestCount = Math.max(
+    ...remaining.map((candidateId) => totalUtility[candidateId]),
+  );
+  remaining = remaining.filter(
+    (candidateId) => totalUtility[candidateId] === bestCount,
+  );
+
+  if (remaining.length === 1) {
+    return remaining;
+  }
+
+  // If still tied, use pre-set tie-break order
+  const orderedRemaining = remaining.sort(
+    (a, b) => tieBreakOrder.indexOf(a) - tieBreakOrder.indexOf(b),
+  );
+
+  return orderedRemaining;
+}
+
+function getUnbreakableTies(
+  candidates: candidate[],
+  cost: Cost,
+  totalUtility: { [candidateId: string]: number },
+): string[][] {
+  const tiedGroups: string[][] = [];
+
+  // Group candidates by cost and initial vote count
+  const groupedByCost = groupBy(candidates, (c) => cost[c.index.toString()]);
+
+  for (const candidateGroup of Object.values(groupedByCost)) {
+    const groupedByVotes = groupBy(
+      candidateGroup,
+      (c) => totalUtility[c.index.toString()],
+    );
+
+    for (const tiedGroup of Object.values(groupedByVotes)) {
+      if (tiedGroup.length > 1) {
+        tiedGroups.push(tiedGroup.map((c) => c.index.toString()));
+      }
+    }
+  }
+
+  return tiedGroups;
+}
+
+function setTieBreakOrder(tiedGroups: string[][]): string[] {
+  const tieBreakOrder: string[] = [];
+
+  for (const tiedGroup of tiedGroups) {
+    // Randomly permute tied group to determine order
+    const groupOrder = shuffle(tiedGroup);
+    tieBreakOrder.push(...groupOrder);
+  }
+
+  return tieBreakOrder;
+}
+
+function shuffle<T>(array: T[]): T[] {
+  const shuffled = array.slice();
+  for (let i = shuffled.length - 1; i > 0; i--) {
+    const j = Math.floor(Math.random() * (i + 1));
```

Comment:
We shouldn't need a random generator here. The "randomTiebreakOrder" input to the main function is intended to determine how "random" ties are broken. This allows tests to be determinate and the server can generate a random order based on candidate UUIDs. Sorry that this wasn't documented.

Reply:
@mikefranze Yeah, makes sense. I'm having issues because the existing tool (https://pref.tools/abcvoting/) doesn't actually describe what order the candidates are elected, so it's hard for me to write tests without me just 'guessing' who should win.